src/core/utils.js

```diff
@@ -3,80 +3,94 @@
 const multihashes = require('multihashes')
 const mapSeries = require('async/mapSeries')
 const CID = require('cids')
+const isIPFS = require('is-ipfs')
 
 exports.OFFLINE_ERROR = 'This command must be run in online mode. Try running \'ipfs daemon\' first.'
 
-const parseIpfsPath = exports.parseIpfsPath = function (pathString) {
-  // example: '/ipfs/b58Hash/links/by/name'
-  //  -> { root: 'b58Hash', links: ['links', 'by', 'name'] }
-  const matched = pathString.match(/^(?:\/ipfs\/)?([^/]+(?:\/[^/]+)*)\/?$/)
-  const errorResult = () => ({
+/**
+ * Break an ipfs-path down into it's root hash and an array of links.
+ *
+ * examples:
+ *  b58Hash -> { root: 'b58Hash', links: [] }
+ *  b58Hash/mercury/venus -> { root: 'b58Hash', links: ['mercury', 'venus']}
+ *  /ipfs/b58Hash/links/by/name -> { root: 'b58Hash', links: ['links', 'by', 'name'] }
+ *
+ * @param  {String} ipfsPath An ipfs-path
+ * @return {Object}            { root: base58 string, links: [string], ?err: Error }
+ */
+exports.parseIpfsPath = function parseIpfsPath (ipfsPath) {
```

Comment:
Validating and decomposing a path into its components seems like a core utility. Do we already have something for this?